scripts/deploy/deploy-webapi.ps1

```diff
@@ -74,25 +91,68 @@ if ($DeploymentSlot) {
-        az webapp deployment slot create --slot $DeploymentSlot --resource-group $ResourceGroupName --name $webappName | Out-Null
+        Write-Host "Deployment slot $DeploymentSlot does not exist, creating..."
+        az webapp deployment slot create --slot $DeploymentSlot --resource-group $ResourceGroupName --name $webApiName --output none
+        $origins = az webapp deployment slot list --resource-group $ResourceGroupName --name $webApiName | ConvertFrom-JSON | Select-Object -Property defaultHostName
     }
 }
 
 Write-Host "Deploying '$PackageFilePath' to Azure WebApp '$webappName'..."
-# Setup the command as a string
-$azWebAppCommand = "az webapp deployment source config-zip --resource-group $ResourceGroupName --name $webappName --src $PackageFilePath"
-
-# Check if DeploymentSlot parameter was passed and append the argument to azwebappcommand
-if ($DeploymentSlot) {
-    $azWebAppCommand += " --slot $DeploymentSlot"
-}
 
 # Invoke the command string
 Invoke-Expression $azWebAppCommand
-
 if ($LASTEXITCODE -ne 0) {
     exit $LASTEXITCODE
 }
+
+if ($EnsureUriInAppRegistration) {
+    $webapiSettings = $(az webapp config appsettings list --name $webapiName --resource-group $ResourceGroupName | ConvertFrom-JSON)
+    $frontendClientId = ($webapiSettings | Where-Object -Property name -EQ -Value Frontend:AadClientId).value
+    $objectId = (az ad app show --id $frontendClientId | ConvertFrom-Json).id
+    $redirectUris = (az rest --method GET --uri "https://graph.microsoft.com/v1.0/applications/$objectId" --headers 'Content-Type=application/json' | ConvertFrom-Json).spa.redirectUris
+    $needToUpdateRegistration = $false
+
+    foreach ($address in $origins) {
+        $origin = "https://$address"
+        Write-Host "Ensuring '$origin' is included in AAD app registration's redirect URIs..."
+
+        if ($redirectUris -notcontains "$origin") {
+            $redirectUris += "$origin"
+            $needToUpdateRegistration = $true
+        }
+    }
+
+    if ($needToUpdateRegistration) {
+        $body = "{spa:{redirectUris:["
+        foreach ($uri in $redirectUris) {
+            $body += "'$uri',"
+        }
+        $body += "]}}"
+
+        az rest `
+            --method PATCH `
+            --uri "https://graph.microsoft.com/v1.0/applications/$objectId" `
+            --headers 'Content-Type=application/json' `
+            --body $body
+        if ($LASTEXITCODE -ne 0) {
+            exit $LASTEXITCODE
+        }
+    }
+}
+
+if ($RegisterPluginCors) {
+    foreach ($pluginName in $pluginNames) {
+        $allowedOrigins = $((az webapp cors show --name $pluginName --resource-group $ResourceGroupName --subscription $Subscription | ConvertFrom-Json).allowedOrigins)
+        foreach ($address in $origins) {
+            $origin = "https://$address"
+            Write-Host "Ensuring '$origin' is included in CORS origins for plugin '$pluginName'..."
+            if (-not $allowedOrigins -contains $origin)) {
```

Comment:
syntax error :/
```suggestion
            if (-not $allowedOrigins -contains $origin) {
```

Reply:
Yes. I've merged a few things in a gung ho manner last night.  Now I'm testing the results and found this bug as well.  Fixed.